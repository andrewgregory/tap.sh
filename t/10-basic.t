#!/usr/bin/bash

dir="$(dirname "$0")"
source "$dir/../tap.sh" || exit 1

runtest() {
    tap_plan 0
    tap_plan 1
    tap_plan 2

    tap_skip_all
    tap_skip_all "foo %s" "bar"

    tap_skip 0
    tap_skip 1 "foo %s" "bar"
    tap_skip 2

    tap_diag "foo"
    tap_diag "foo %s" "bar"

    tap_bail
    tap_bail "foo %s" "bar"

    tap_ok 1
    tap_ok 0
    tap_ok 1 "%s" "foo"
    tap_ok 0 "%s" "foo"

    tap_is_str "foo" "foo"
    tap_is_str "foo" "bar"
    tap_is_str "foo" "foo" "foo %s" "bar"
    tap_is_str "foo" "bar" "foo %s" "bar"
    tap_is_str "" ""
    tap_is_str "" "bar"

    tap_is_int 1 1
    tap_is_int 1 0 
    tap_is_int 1 1 "foo %s" "bar"
    tap_is_int 1 0 "foo %s" "bar"

    tap_diff <(printf "1\n2\n3\n") <(printf "1\n2\n3\n") "tap_diff"
    tap_diff <(printf "1\n2\n3\n") <(printf "3\n2\n1\n") "tap_diff"

    tap_like "foo"    "^\<(foo|bar)\>$"
    tap_like "bar"    "^\<(foo|bar)\>$"
    tap_like "foobar" "^\<(foo|bar)\>$"
    tap_like "foo"    "^\<(foo|bar)\>$" "foo %s" "bar"
    tap_like "bar"    "^\<(foo|bar)\>$" "foo %s" "bar"
    tap_like "foobar" "^\<(foo|bar)\>$" "foo %s" "bar"
    tap_like "" ""
    tap_like "" "^\<(foo|bar)\>"

    tap_done_testing
}

run() {
    runtest
    tap_todo="FOO TODO"
    runtest
}

output="$(diff -u --label got --label expected <(run 2>&1) "$dir/expected/10-basic.t.out")"
ret=$?

printf "1..1\n"
if [[ $ret -eq 0 ]]; then
    printf "ok 1\n"
    exit 0
else
    printf "not ok 1\n"
    sed -e 's/^/   # /' <<<"$output"
    exit 1
fi

# vim: ft=sh
