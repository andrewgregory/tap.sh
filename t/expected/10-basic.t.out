1..0
1..1
1..2
1..0 # SKIP
1..0 # SKIP foo bar
ok 1 # SKIP foo bar
ok 2 # SKIP
ok 3 # SKIP
# foo
# foo bar
Bail out!
Bail out! foo bar
not ok 4
#   Failed test at 10-basic.t line 24.
ok 5
not ok 6 - foo
#   Failed test at 10-basic.t line 26.
ok 7 - foo
ok 8
not ok 9
#   Failed test at 10-basic.t line 30.
#          got: 'foo'
#     expected: 'bar'
ok 10 - foo bar
not ok 11 - foo bar
#   Failed test at 10-basic.t line 32.
#          got: 'foo'
#     expected: 'bar'
ok 12
not ok 13
#   Failed test at 10-basic.t line 34.
#          got: ''
#     expected: 'bar'
ok 14
not ok 15
#   Failed test at 10-basic.t line 37.
#          got: '1'
#     expected: '0'
ok 16 - foo bar
not ok 17 - foo bar
#   Failed test at 10-basic.t line 39.
#          got: '1'
#     expected: '0'
ok 18 - tap_diff
not ok 19 - tap_diff
#   Failed test at 10-basic.t line 42.
# --- got
# +++ expected
# @@ -1,3 +1,3 @@
# -1
# -2
#  3
# +2
# +1
ok 20
ok 21
not ok 22
#   Failed test at 10-basic.t line 46.
#          got: 'foobar'
#     expected: /^\<(foo|bar)\>$/
ok 23 - foo bar
ok 24 - foo bar
not ok 25 - foo bar
#   Failed test at 10-basic.t line 49.
#          got: 'foobar'
#     expected: /^\<(foo|bar)\>$/
ok 26
not ok 27
#   Failed test at 10-basic.t line 51.
#          got: ''
#     expected: /^\<(foo|bar)\>/
1..27
1..0
1..1
1..2
1..0 # SKIP
1..0 # SKIP foo bar
ok 28 # SKIP foo bar
ok 29 # SKIP
ok 30 # SKIP
# foo
# foo bar
Bail out!
Bail out! foo bar
not ok 31 # TODO FOO TODO
#   Failed (TODO) test at 10-basic.t line 24.
ok 32 # TODO FOO TODO
not ok 33 - foo # TODO FOO TODO
#   Failed (TODO) test at 10-basic.t line 26.
ok 34 - foo # TODO FOO TODO
ok 35 # TODO FOO TODO
not ok 36 # TODO FOO TODO
#   Failed (TODO) test at 10-basic.t line 30.
#          got: 'foo'
#     expected: 'bar'
ok 37 - foo bar # TODO FOO TODO
not ok 38 - foo bar # TODO FOO TODO
#   Failed (TODO) test at 10-basic.t line 32.
#          got: 'foo'
#     expected: 'bar'
ok 39 # TODO FOO TODO
not ok 40 # TODO FOO TODO
#   Failed (TODO) test at 10-basic.t line 34.
#          got: ''
#     expected: 'bar'
ok 41 # TODO FOO TODO
not ok 42 # TODO FOO TODO
#   Failed (TODO) test at 10-basic.t line 37.
#          got: '1'
#     expected: '0'
ok 43 - foo bar # TODO FOO TODO
not ok 44 - foo bar # TODO FOO TODO
#   Failed (TODO) test at 10-basic.t line 39.
#          got: '1'
#     expected: '0'
ok 45 - tap_diff # TODO FOO TODO
not ok 46 - tap_diff # TODO FOO TODO
#   Failed (TODO) test at 10-basic.t line 42.
# --- got
# +++ expected
# @@ -1,3 +1,3 @@
# -1
# -2
#  3
# +2
# +1
ok 47 # TODO FOO TODO
ok 48 # TODO FOO TODO
not ok 49 # TODO FOO TODO
#   Failed (TODO) test at 10-basic.t line 46.
#          got: 'foobar'
#     expected: /^\<(foo|bar)\>$/
ok 50 - foo bar # TODO FOO TODO
ok 51 - foo bar # TODO FOO TODO
not ok 52 - foo bar # TODO FOO TODO
#   Failed (TODO) test at 10-basic.t line 49.
#          got: 'foobar'
#     expected: /^\<(foo|bar)\>$/
ok 53 # TODO FOO TODO
not ok 54 # TODO FOO TODO
#   Failed (TODO) test at 10-basic.t line 51.
#          got: ''
#     expected: /^\<(foo|bar)\>/
1..54
